src/Components/Blazor/WebAssembly.Authentication/test/RemoteAuthenticationServiceTests.cs

```diff
@@ -0,0 +1,492 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Microsoft.JSInterop;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    public class RemoteAuthenticationServiceTests
+    {
+        [Fact]
+        public async Task RemoteAuthenticationService_SignIn_UpdatesUserOnSuccess()
+        {
+            // Arrange
+            var testJsRuntime = new TestJsRuntime();
+            var options = CreateOptions();
+            var runtime = new RemoteAuthenticationService<RemoteAuthenticationState, OidcProviderOptions>(
+                testJsRuntime,
+                options);
+
+            var state = new RemoteAuthenticationState();
+            testJsRuntime.SignInResult = new RemoteAuthenticationResult<RemoteAuthenticationState>
+            {
+                State = state,
+                Status = RemoteAuthenticationStatus.Success
+            };
+
+            // Act
+            await runtime.SignInAsync(new RemoteAuthenticationContext<RemoteAuthenticationState> { State = state });
+
+            // Assert
+            Assert.Equal(
+                new[] { "AuthenticationService.init", "AuthenticationService.signIn", "AuthenticationService.getUser" },
+                testJsRuntime.PastInvocations.Select(i => i.identifier).ToArray());
```

Comment:
The test shows we're making these JS interop calls, but doesn't appear to show that the authentication status was changed on the .NET side.

Reply:
We have coverage for that in our E2E tests. I might look to add more tests if I see a gap there.

Reply:
It is not obvious, but the call to `AuthenticationService.getUser` implies that the user is being updated.

I have other tests below where I test the GetAuthenticatedUser logic specifically, so I'm going to resolve this.